lemur/certificates/service.py

```diff
@@ -560,18 +560,35 @@ def query_common_name(common_name, args):
     :return:
     """
     owner = args.pop("owner")
-    if not owner:
-        owner = "%"
-
     # only not expired certificates
     current_time = arrow.utcnow()
 
-    result = (
-        Certificate.query.filter(Certificate.cn.ilike(common_name))
-        .filter(Certificate.owner.ilike(owner))
-        .filter(Certificate.not_after >= current_time.format("YYYY-MM-DD"))
-        .all()
-    )
+    if common_name == "%" and not owner:
+        result = (
+            Certificate.query.filter(Certificate.not_after >= current_time.format("YYYY-MM-DD"))
+            .filter(not_(Certificate.revoked))
+            .filter(not_(Certificate.replaced.any()))  # ignore rotated certificates to avoid duplicates
+            .all()
+        )
+    elif common_name == "%":
+        # all valid certs from the owner
+        result = (
+            Certificate.query.filter(Certificate.not_after >= current_time.format("YYYY-MM-DD"))
+            .filter(Certificate.owner == owner)
+            .filter(not_(Certificate.revoked))
+            .filter(not_(Certificate.replaced.any()))  # ignore rotated certificates to avoid duplicates
+            .all()
+        )
+    else:
+        # search based on owner and cn
+        result = (
+            Certificate.query.filter(Certificate.not_after >= current_time.format("YYYY-MM-DD"))
+            .filter(Certificate.cn.like(common_name))
```

Comment:
The only difference between like and ilike is that ilike is case insensitive.  So in this case, if common_name=FOO.net and Certificate.cn = foo.net, there will be not be a match.  Not sure if that is an issue, but it might be.  Might be better to keep it as ilike to meet expectations.

Reply:
done